checker/src/abstract_value.rs

```diff
@@ -2565,8 +2565,14 @@ impl AbstractValueTrait for Rc<AbstractValue> {
             | Expression::HeapBlockLayout { .. }
             | Expression::Reference { .. }
```

Comment:
If the referenced value has been tagged, we would like to see here.

Reply:
Similar to the issue on tag checks: can reference propagate tags?

Reply:
I can't quite see why I would like a reference to hide the tags of the value being referred to.

Reply:
My assumption there was that heap addresses do not lack information. We can definitely make references propagate everything, which is safe and makes tag checks easier.

Reply:
Well, this does not really make tag checks easier. If references propagate tags, then dereferences should also propagate them for the same reason. If this is the case, then we might need to check `&&&&&...secret` when we just want to check `secret`.

Reply:
Perhaps this is a non issue. Whenever we check for the presence or absence of a tag, the reference will get dereferenced before the tags domain is accessed.